### ISSUE TITLE
WFS as datasource - getting Timeout waiting for connection from pool

### ISSUE THREAD

**Issue by hesmebye** [interactive-instruments/ldproxy#774]
### ldproxy Version

3.2.4

### Current Behavior

I am getting 

  "status" : 400,
  "title" : "Bad Request",
  "detail" : "org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool"

When calling the service on every X request.  Makes it random which features thats missing.

### Expected Behavior

According to a thread on stackechange:

return connection back to pool, so that the abowe message does not occur

### Steps to Reproduce

Multiple simulataniously requests fetching polygons (as I did)

### Relevant log output

```shell
org.apache.http.conn.ConnectionPoolTimeoutException: Timeout waiting for connection from pool
```

**Comment by azahnen** [comment]
Thanks for the catch! It indeed looks like there is a connection leak, I will look into it in the next days.

**Comment by azahnen** [comment]
After more thorough testing, I could rule out a connection leak, but I could pretty easily trigger the exception with a load test. It turned out the defaults for max connections per route as well as pool timeouts are pretty small. Fortunately, you can adjust these settings in the `cfg.yml`. These would be the relevant default settings:

```
httpClient:
  connectionRequestTimeout: 500ms
  maxConnectionsPerRoute: 16
  maxConnections: 1024
```

I would suggest to increase both `connectionRequestTimeout ` and `maxConnectionsPerRoute `. Let me know if that works for you and good luck with the presentation!

**Comment by hesmebye** [comment]
Thanks - I will test that :)

**Comment by hesmebye** [comment]
> After more thorough testing, I could rule out a connection leak, but I could pretty easily trigger the exception with a load test. It turned out the defaults for max connections per route as well as pool timeouts are pretty small. Fortunately, you can adjust these settings in the `cfg.yml`. These would be the relevant default settings:
> 
> ```
> httpClient:
>   connectionRequestTimeout: 500ms
>   maxConnectionsPerRoute: 16
>   maxConnections: 1024
> ```
> 
> I would suggest to increase both `connectionRequestTimeout ` and `maxConnectionsPerRoute `. Let me know if that works for you and good luck with the presentation!

I have been searching the documentation without finding where to find the cfg.yaml file.  I have placed it under the data folder.  Is that correct?

**Comment by azahnen** [comment]
That is correct, but it has to be `cfg.yml` (We plan to improve the docs drastically by the end of the year, currently it is only shortly mentioned [here](https://docs.ldproxy.net/application/30-data-folder.html)). 

You might set the [log level](https://docs.ldproxy.net/application/50-logging.html#log-level) to `DEBUG` as an easy way to validate that the `cfg.yml` is actually loaded.

**Comment by hesmebye** [comment]
> That is correct, but it has to be `cfg.yml` (We plan to improve the docs drastically by the end of the year, currently it is only shortly mentioned [here](https://docs.ldproxy.net/application/30-data-folder.html)).
> 
> You might set the [log level](https://docs.ldproxy.net/application/50-logging.html#log-level) to `DEBUG` as an easy way to validate that the `cfg.yml` is actually loaded.

Thanks - will test now!

**Comment by azahnen** [comment]
I hope the proposed solution worked for you, I will close this issue.
